### DISCUSSION TITLE
How do you know when the companion app has been updated?

### DISCUSSION THREAD

**Discussion by nollinvoyd** [aclap-dev/video-downloadhelper#74] [Help and Q&A]
### Your question

I came seeking info on something else, and realized there was a newer version of the companion app available.

Is there a alert mechanism for informing users?

**Comment by paulrouget** [comment]
The companion app doesn't need to be updated. If VDH needs a feature or a bug fix that is not in the installed CoApp, VDH will show a message and a link to install the latest CoApp.

Most of the time, the new versions just fix CoApp installation issues.

**Reply by nollinvoyd** [reply]
Please reread my original message.

I said when I came here today, while reading some of the threads, I realized that my v2.0.10 of the companion app was out of date, after which I discovered there was an update to v2.0.17.  

My question was, when there is a companion app update, is there some established method in place to inform users of an update? Some type of notification? 

"If VDH needs a feature or a bug fix that is not in the installed CoApp, VDH will show a message and a link to install the latest CoApp".  

     Is there a setting that must be activated for that?  I received no such message.

**Reply by GrampaWildWilly** [reply]
Are you asking for the CoApp installer page to show the version number of the latest release?

**Reply by nollinvoyd** [reply]
No.

I said, I would never have known there "was" an update if I hadn't , just by chance, discovered when I came here that there was an update to be had.

paulrouget said "VDH will show a message and a link to install the latest CoApp".  

I have never seen a message resembling that.  I don't want to be using an older version that will compromise the operation.

**Reply by GrampaWildWilly** [reply]
I believe, again I'm not a dev, that the latest public release of VDH is set to ask for a new CoApp only when the public release is meant to ask for one.  If you install a new public release, it will most likely ask for a newer CoApp.  But I admit I am guessing.  There has been a series of new beta versions of the browser add-on.  I suspect each one of those would ask for a newer CoApp.  I am not certain about that because I've been manually updating the CoApp probably before I let any of the beta add-ons ask for one.

But if you're not asking for the release numbers on a web page, I am.  Let me give some examples.

https://github.com/sabrogden/Ditto Another product hosted here on GitHub.  A short way down that page on the right there is a Releases link showing the latest release number & its release date.  This is the usual format of home pages on GitHub.  The VLC & CoApp pages on GitHub are much like this, except for the VLC page failing to show a latest release number & release date.

https://www.codecguide.com/klcp_update.htm Latest release number & release date are shown near the top of the page.

https://notepad-plus-plus.org/ Latest release number but not the release date are shown near the top of the page.  But if you click the latest release number on that page, it takes you to a page where the latest release date is prominently displayed.

https://www.faststone.org/ Latest release number & release date are shown near the top of the page.  That page contains information (release numbers & release dates) for several products, but I have only the one at the top.

http://nikse.dk/ Latest release number & release date are shown near the top of the page.  This product is also hosted on GitHub, including its very active support forum.  There is a nearly continuous stream of beta updates available: https://github.com/SubtitleEdit/subtitleedit/releases.  Although it doesn't show the subrelease number for the beta, it does show the date on which it was last updated, which serves the purpose.  If you hover your mouse over the date on the right, it floats up a tooltip showing the exact date & time of last update.

https://www.openoffice.org/ Latest release number is prominently displayed at the top of the page.  The latest release date is easily found just a short way down the page.

http://www.videolan.org/ Latest release number & release date are shown near the top of the page.

Firefox is slightly, but only slightly, less straightforward.  You visit this page: https://www.mozilla.org/en-US/.  I'm sure there are home pages for other regions & languages that must be similar.  At the top of the page, there is a button labelled Firefox Browsers you hover your mouse over.  It drops down a menu of choices, one of which is Release Notes.  When you click that, it shows you the release number of the most recent release & the release date.

VDH/CoApp needs to adhere to what is obviously a widely-accepted best practice.  I feel compelled to point out that software in general, VDH/CoApp in particular, is not here for the convenience of the developers.  Software is here for the convenience of the users.  If something is inconvenient for the developers, tough, that does not matter in the slightest.  Only the convenience of the users is of any concern.  Clearly, release number & release date are for the convenience of us users.  They should be clearly displayed on these pages:

https://github.com/aclap-dev/video-downloadhelper/wiki
https://github.com/aclap-dev/video-downloadhelper/wiki/Beta-versions
https://github.com/aclap-dev/video-downloadhelper/wiki/CoApp-Installation

And honestly, how long does it take to change the release number & release date on a web page each time you update the product?  I say inconvenience for the developers is a bogus claim.  Just do it.

**Comment by nollinvoyd** [comment]
deleted

**Comment by HereButNotaGit** [comment]
I came here wondering why there was no version number cited on the download page for the CoApp on the VDH website.  I have a multitab bookmark on my browser for key applications and browser extensions, and I launch that tab daily and flick through download pages for various things to check proactively if any of those bookmarked pages have changed, so that I can detect software updates.  I also make a practice of downloading and keeping full offline installers for anything that I can because sometimes updates include bugs that can only be avoided by reinstalling a previous version of the software.  I had occasion to do such a backlevelling as recently as yesterday.

While doing such a routine daily check for possible updates, I saw that the CoApp download page had changed from the one that explicitly announced version 2.0.10, but it wasn't clear whether both the CoApp and download page had been updated, or just the page.  In the end, I downloaded the CoApp installer file, saw that it was different from the 2.0.10 installer file, installed the new CoApp, found from Windows Control Panel that it was version 2.0.17, and then changed the versionless name of the installer file to include the version number.

Paul, for version 2.0.17 of the CoApp, why did you not include the version number on the download page and in the name of the installer file?  It's not supposed to be a detective hunt to determine the version of the software that is being offered for download.

While on the topic of the download page for the CoApp, is there anywhere on the VDH website that links to the CoApp download page?  I've already had it bookmarked for a long time, but could only find the URL from a Google search.  I'm thinking of newbies to the VDH website wondering where to find the CoApp.

**Comment by paulrouget** [comment]
I've added the current release number to the download page.

**Reply by GrampaWildWilly** [reply]
I looked at the 3 download pages I mentioned above.  Not one of them shows a release number.  I did Ctrl+F5 on each of them, which is supposed to bypass my browser disk cache.  Is this another of those cases of web pages needing to be flushed from web cache?

**Reply by paulrouget** [reply]
ah, CDN. My bad.

**Reply by GrampaWildWilly** [reply]
Yep.  It's there on the third URL I quoted above.  This is good.
